### ISSUE TITLE
CODE: 10007  DESC: create reply message fail, requestMessage error, property[CLUSTER] is null.

### ISSUE THREAD

**Issue by zmapleshine** [apache/rocketmq-spring#243]
The issue tracker is **ONLY** used for bug report and feature request. 

Any question or RocketMQ proposal please use our [mailing lists](http://rocketmq.apache.org/about/contact/).

**BUG REPORT**

1. Please describe the issue you observed:

- What did you do (The steps to reproduce)?
  download this example and run producer and consumer project.
- What did you expect to see?
   no error 
- What did you see instead?
    consume message failed .
2. Please tell us about your environment:
    java 8 win10 IDEA 2019.3,
    rocketMQ deployed docker container, include : 
        one cluster: DefaultCluster
        two broker: each of them contains one slave.
       list the four configuration files:

    brokerClusterName=DefaultCluster
    brokerName=broker-main
    brokerId=0

    brokerClusterName=DefaultCluster
    brokerName=broker-main
    brokerId=1

    brokerClusterName=DefaultCluster
    brokerName=brokerA
    brokerId=0

    brokerClusterName=DefaultCluster
    brokerName=brokerA
    brokerId=1

3. Other information (e.g. detailed explanation, logs, related issues, suggestions how to fix, etc):

`
2020-03-25 16:27:36.733  WARN 40768 --- [essageThread_10] a.r.s.s.DefaultRocketMQListenerContainer : consume message failed. messageExt:MessageExt [queueId=1, storeSize=406, queueOffset=0, sysFlag=0, bornTimestamp=1585124856698, bornHost=/10.176.200.23:60310, storeTimestamp=1585124670921, storeHost=/10.134.132.11:10911, msgId=0A86840B00002A9F0000000000149542, commitLogOffset=1348930, bodyCRC=1338495376, reconsumeTimes=0, preparedTransactionOffset=0, toString()=Message{topic='stringRequestTopic', flag=0, properties={MIN_OFFSET=0, REPLY_TO_CLIENT=10.0.75.1@40480, MAX_OFFSET=1, CONSUME_START_TIME=1585124856733, id=dd02d8b0-6ba7-aa7d-45f9-f74335ddd19f, UNIQ_KEY=20010000348BFB582C44C042487C8E71000018B4AAC27F20CF7A0055, WAIT=false, contentType=text/plain;charset=UTF-8, TTL=300000, TAGS=tagA, timestamp=1585124856598, CORRELATION_ID=8028d503-ca08-4366-8965-a950d3369030}, body=[114, 101, 113, 117, 101, 115, 116, 32, 115, 116, 114, 105, 110, 103], transactionId='null'}], error:{}

org.apache.rocketmq.client.exception.MQClientException: CODE: 10007  DESC: create reply message fail, requestMessage error, property[CLUSTER] is null.
For more information, please visit the url, http://rocketmq.apache.org/docs/faq/
	at org.apache.rocketmq.client.utils.MessageUtil.createReplyMessage(MessageUtil.java:46) ~[rocketmq-client-4.6.0.jar:4.6.0]
	at org.apache.rocketmq.spring.support.DefaultRocketMQListenerContainer.handleMessage(DefaultRocketMQListenerContainer.java:399) ~[rocketmq-spring-boot-2.1.0.jar:2.1.0]
	at org.apache.rocketmq.spring.support.DefaultRocketMQListenerContainer.access$100(DefaultRocketMQListenerContainer.java:70) ~[rocketmq-spring-boot-2.1.0.jar:2.1.0]
	at org.apache.rocketmq.spring.support.DefaultRocketMQListenerContainer$DefaultMessageListenerConcurrently.consumeMessage(DefaultRocketMQListenerContainer.java:354) ~[rocketmq-spring-boot-2.1.0.jar:2.1.0]
	at org.apache.rocketmq.client.impl.consumer.ConsumeMessageConcurrentlyService$ConsumeRequest.run(ConsumeMessageConcurrentlyService.java:411) [rocketmq-client-4.6.0.jar:4.6.0]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [na:1.8.0_231]
	at java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:266) [na:1.8.0_231]
	at java.util.concurrent.FutureTask.run(FutureTask.java) [na:1.8.0_231]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [na:1.8.0_231]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [na:1.8.0_231]
	at java.lang.Thread.run(Thread.java:748) [na:1.8.0_231]
`

**Comment by zmapleshine** [comment]
@RongtongJin

**Comment by RongtongJin** [comment]
Hi, @zmapleshine No such issue in my environment. @xiangwangcheng Could you help verify if there is such issue?

**Comment by xiangwangcheng** [comment]
OK. I will try it. @zmapleshine @RongtongJin

**Comment by zmapleshine** [comment]
Now I have solved this problem and it has something to do with the cluster environment configuration.

At the beginning, I traced the source code of "rocketmq-client" and found that there was no place to set the property "CLUSTER", it was only set in "rocketmq-all", but the official "rocketmq-spring-boot-starter" did not include this jar .

Because there is no clue, I carefully checked the steps of server setup, and found through the "web console" that the "cluster" list shows only two records. The reason is that there is a problem with the port mapping of docker. 

Second, I installed the latest version 4.7.0, but the latest "Rocketmq-starter" contains the client jar version 4.6.0. 

Although I did not verify whether the different versions of the client and server can work normally, I still downgraded the server version to 4.6.0. Then this problem did not occur.

 It is just strange that the property "CLUSTER" located in rocketmq-client has not been set artificially from beginning to end. Where is it set?...
